File: HarvardX-CS50-project03-Cash/Python/cash.py
def main():
    change_owed = 0

    # validate input
    while True:
        change_owed = input("Change owed: ")
        if is_float(change_owed):
            change_owed = float(change_owed)
            if change_owed > 0:
                break

    # init variables
    cents = round(change_owed * 100)
    coin = (25, 10, 5, 1)

    print(count(cents, coin))

# functions definitions


def count(cents, coin):
    current_coin = 0
    total_coins = 0
    reminder = 0

    while current_coin < len(coin):

        reminder = cents % coin[current_coin]

        if reminder >= 0:
            cents -= reminder
            total_coins += (cents // coin[current_coin])
            cents = reminder
        current_coin += 1

    return total_coins


def is_float(num):
    try:
        num = float(num)
        return True
    except ValueError:
        return False


if __name__ == "__main__":
    main()

File: HarvardX-CS50-project03-Cash/C/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int cents;
int total_coins;

void count(int,int);

int main(void)
{
    float change_owed;
    do
    {
        change_owed = get_float("Change owed: ");
    }
    while(change_owed < 0.00);
    
    cents = round(change_owed*100);
    total_coins = 0;

    int quarters = 25;
    int dimes = 10;
    int nickles = 5;
    int pennies = 1;
    int remainder = 0;
    
    
    while(cents > 0)
    {
        while(cents >= quarters)
        {
            count(quarters, remainder);          
        } 
        while(cents >= dimes)
        {
            count(dimes, remainder); 
        }
        while(cents >= nickles)
        {
            count(nickles, remainder); 
        }
        while(cents >= pennies)
        {
            count(pennies, remainder); 
        }
    }
    printf("%i\n", total_coins);
}

void count(int value, int remainder)
{
    remainder =  cents / value;
    total_coins += remainder;
    cents -= value * remainder;
}

